docs/reference.rst

```diff
@@ -1027,36 +1050,72 @@ In future releases, they will be accessible in other hooks, as well.
 
 .. _sec-ref-supported-image-types:
 
-Supported Image Types (Extensions)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-RAUC uses the file name extension of images in combination with the target
-slot's type to select the *update handler*.
-To keep the number of combinations manageable, only a limited amount of file
-name extensions is supported by RAUC.
+Supported Image Types
+~~~~~~~~~~~~~~~~~~~~~
+.. note::
+  RAUC v1.15 or newer support setting an explicit ``type`` field
+  instead of using the filename extensions to determine the image type.
+  For backward compatibility, RAUC can still derive the image type
+  from file name extensions.
 
-The ``*.img`` extension is valid for all slot types.
-It should be used when no specific file name extension (and handler) is
-supported.
+RAUC uses the image ``type`` in combination with the slot ``type``
+to determine how to process images during installation.
+When the ``type`` parameter is set, the filename extension is not taken into
+account and can be chosen freely.
 
-Supported file system image types/extensions are:
+Supported file system image types are:
 
-  * ``*.ext4``: ext[234] file system image
-  * ``*.vfat``: vfat/EFI file system image
-  * ``*.ubifs``: UBIFS file system image
-  * ``*.squashfs``: SquashFS image (compressed or uncompressed)
-  * ``*.squashfs-<comp>``:  SquashFS image (compressed, with ``<comp>`` being one of ``lz4``, ``lzo``, ``xz``, ``xst``)
+  * ``ext4``: ext[234] file system image
+  * ``vfat``: vfat file system image (sometimes used for the EFI System Partition)
+  * ``ubifs``: UBIFS file system image
+  * ``squashfs``: SquashFS image (uncompressed, or compressed with one of ``lz4``, ``lzo``, ``xz``, ``xst``)
+  * ``emptyfs``: Creates an empty filesystem.
+  This type does not support setting a `filename` as no source data is needed.
```

Comment:
The indentation looks wrong here if the sentence belongs to `emptyfs` image type.